libcxx/include/string

```diff
@@ -1198,11 +1198,17 @@ public:
 
   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX20 const_reference operator[](size_type __pos) const _NOEXCEPT {
```

Comment:
Let's add a few tests with clearly-constant indices in `libcxx/test/std/strings/basic.string/string.access/index.pass.cpp`. Otherwise I think this code path might not be tested.

Reply:
Done